src/portal_visualization/builders/anndata_builders.py

```diff
@@ -122,11 +166,16 @@ def _setup_anndata_view_config(self, vc, dataset, marker=None):
         # Vitessce feature request to add it:
         # https://github.com/vitessce/vitessce/issues/1298
 
-        self._add_spatial_view(dataset, vc)
+        # Spatial view is added if present, otherwise gets filtered out before views are linked
+        # This ensures that the view config is valid for datasets with and without a spatial view
+        spatial = self._add_spatial_view(dataset, vc)
+
+        views = [v for v in [cell_sets, gene_list, scatterplot,
```

Comment:
https://docs.python.org/3/library/functions.html#filter ?

Reply:
`filter` should work here as well - I reached for list comprehension as the tool I'm more used to from previous Python experience, but we can use `filter` if that's the preferred solution:

```
       views = list(filter(lambda v: v is not None, [cell_sets, gene_list, scatterplot, cell_sets_expr, heatmap, spatial]))
```